routes/auth_routes.js

```diff
@@ -0,0 +1,49 @@
+var express = require('express');
+var https = require('https');
+var clientId = process.env.CLIENT_ID;
+var clientSecret = process.env.CLIENT_SECRET;
+
+var authRouter = module.exports = express.Router();
+
+authRouter.get('/', function(req, res) {
+  res.redirect('https://github.com/login/oauth/authorize/?client_id=' + clientId);
+});
+
+authRouter.get('/token', function(req, res) {
+  var options = {
+    host: 'github.com',
+    path: '/login/oauth/access_token?client_id=' +
+      clientId + '&client_secret=' + clientSecret + '&code=' +
+      req.query.code,
+    method: 'POST',
+    headers: {
+      Accept: 'application/json'
+    }
+  };
+  var postData = JSON.stringify({
+    client_id: clientId,
+    client_secret: clientSecret,
+    code: req.query.code
+  });
+  var sendOauth = https.request(options, function(authRes) {
+    var data = '';
+    authRes.setEncoding('utf8');
+    authRes.on('data', function(chunk) {
+      data += chunk;
+    });
+    authRes.on('end', function() {
+      req.oauth = JSON.parse(data);
+      // check for errors, store token and redirect
+
```

Comment:
to be honest, you're going to have a much easier time using the [supergent](https://github.com/visionmedia/superagent) library than trying to do this with the vanilla http stuff, but basically you should get the token back, store it in the database for the user and generate a token (jwt or eat, doesn't matter, probably should go with jwt though) and send it back. Remember this is an api, not a server side rendered web app. We can talk about it sometime tomorrow(Thursday).